cura/Machines/MaterialManager.py

```diff
@@ -0,0 +1,324 @@
+from collections import defaultdict, OrderedDict
+from typing import Optional
+
+from PyQt5.Qt import QTimer, QObject, pyqtSignal
+
+from UM.Logger import Logger
+from UM.Settings import ContainerRegistry
+
+from cura.Machines.ContainerNode import ContainerNode
+
+
+class MaterialGroup:
```

Comment:
Best document this. What sort of group is this supposed to represent? A group of materials that is derived from the same base file?

Reply:
Added documentation for `MaterialGroup` in c5d443109